packages/components/modules/profiles/web/ProfileComponent/BlockButtonWithDialog/index.tsx

```diff
@@ -71,26 +71,44 @@ const BlockButtonWithDialog: FC<BlockButtonWithDialogProps> = ({
 
   return (
     <>
-      <Button
-        variant={isMenu ? 'text' : 'contained'}
-        onClick={handleOpen}
-        sx={{ justifyContent: isMenu ? 'start' : 'center' }}
-        size="medium"
-      >
-        <Typography variant="body2" color={isMenu ? 'error.main' : 'inherit'} noWrap>
-          {isBlockedByMe ? (
-            <>
-              <UnblockIcon sx={{ color: isMenu ? 'error.main' : 'inherit', marginRight: '5px' }} />
-              {isMenu ? 'Unblock profile' : 'Unblock'}
-            </>
-          ) : (
-            <>
-              <BlockIcon sx={{ color: 'error.main', marginRight: '5px' }} />
-              Block profile
-            </>
-          )}
-        </Typography>
-      </Button>
+      {isMenu ? (
+        <MenuItem onClick={handleOpen} disableRipple>
+          <Typography variant="body2" color="error.main" noWrap>
+            {isBlockedByMe ? (
+              <>
+                <UnblockIcon sx={{ color: 'error.main', marginRight: '5px' }} />
+                Unblock profile
+              </>
+            ) : (
+              <>
+                <BlockIcon sx={{ color: 'error.main', marginRight: '5px' }} />
+                Block profile
+              </>
+            )}
+          </Typography>
+        </MenuItem>
+      ) : (
+        <Button
+          variant="contained"
+          onClick={handleOpen}
+          sx={{ justifyContent: 'center' }}
+          size="medium"
+        >
+          <Typography variant="body2" color="inherit" noWrap>
+            {isBlockedByMe ? (
+              <>
+                <UnblockIcon sx={{ color: 'inherit', marginRight: '5px' }} />
+                Unblock
```

Comment:
Should this be "Unblock Profile" like in the Menu one?
Also, on both, the MenuItem and the Button, the code is basically the same. Can we make this a separate component?

Reply:
I don't know what this should be as it was unrelated to my story, I just adapted it so it's more aligned with the rest. If you want to, I can change, but please confirm that it's necessary